src/main/java/com/google/vinet/data/MatchingRunner.java

```diff
@@ -0,0 +1,194 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.vinet.data;
+
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.*;
+
+public class MatchingRunner {
+  private DatastoreService datastore;
+  private Set<IsolateTimeSlot> isolateTimeSlots;
+  private Set<VolunteerTimeSlot> volunteerTimeSlots;
+  private LocalDate date;
+
+  /**
+   * Construct a MatchingRunner with none of its data set.
+   */
+  public MatchingRunner() {
+
+  }
+
+  /**
+   * Construct a matching runner with a dependency on the provided DataStore implementation.
+   * @param datastore The DataStore implementation that this MatchingRunnner will depend on.
+   */
+  protected MatchingRunner(DatastoreService datastore) {
+    Objects.requireNonNull(datastore);
+    this.datastore = datastore;
+  }
+
+  /**
+   * Run the matching algorithm and store the results in DataStore.
+   * Any necessary data that is not already set will be pulled from DataStore, with an assumption
+   * that matches should be created that will be scheduled for tomorrow.
+   */
+  public void run() {
+    if (datastore == null) datastore = DatastoreServiceFactory.getDatastoreService();
+    if (date == null) date = LocalDate.now().plusDays(1);
+    if (isolateTimeSlots == null) isolateTimeSlots = this.fetchIsolateTimeSlots();
+    if (volunteerTimeSlots == null) volunteerTimeSlots = this.fetchVolunteerTimeSlots();
+
+    deletePreviousMatches(date, datastore);
+
+    final Set<IsolateTimeSlot> matches = MatchingAlgorithm.matchTimeSlots(isolateTimeSlots, volunteerTimeSlots);
+
+    for (IsolateTimeSlot matching : matches) {
+      Entity matchingEntity = new Entity("Matching");
+      matchingEntity.setProperty("isolateId", matching.getIsolate().userId);
+      matchingEntity.setProperty("volunteerId", matching.getPairedSlot().registeredUser.userId);
+      matchingEntity.setProperty("date", matching.date.toString());
+      matchingEntity.setProperty("start", matching.start.toString());
+      matchingEntity.setProperty("end", matching.end.toString());
+      matchingEntity.setProperty("ticket", KeyFactory.keyToString(matching.ticket));
+      datastore.put(matchingEntity);
+    }
+  }
+
+  public static void deletePreviousMatches(LocalDate date, DatastoreService datastore){
```

Comment:
Does this need to be a public function?

Reply:
This has been fixed now, it should have been protected: https://github.com/googleinterns/step226-2020/pull/10/commits/3017562df5ae01c1842569ee3773a973dbaf8d3c